### PR TITLE
Fix build with EXTRA_METADATA enabled / Update dirty flag

### PR DESCRIPTION
**Pull request by uklotzde**
@daschuer The EXTRA_METADATA feature is broken, caused by #2645.

**Update**: I might have found a minor bug. The dirty flag was not updated when merging re-imported metadata. I also decided to update the documentation, because I got confused by what I had written originally :see_no_evil:

### CODE REVIEW

**Comment by daschuer** [comment]
Sorry for eating my dog food :-/
Where do you usually define __EXTRA_METADATA__? For Testing I have put it into util/macros.h Just building.

File: src/track/trackrecord.cpp [hunk 1]
```diff
@@ -102,37 +115,39 @@ void TrackRecord::mergeImportedMetadata(
...
+    modified |= copyIfNotNull(pMergedTrackInfo->ptrDiscNumber(), importedTrackInfo.getDiscNumber());
+    modified |= copyIfNotNull(pMergedTrackInfo->ptrDiscTotal(), importedTrackInfo.getDiscTotal());
+    modified |= copyIfNotNull(pMergedTrackInfo->ptrEncoder(), importedTrackInfo.getEncoder());
+    modified |= copyIfNotNull(pMergedTrackInfo->ptrEncoderSettings(), importedTrackInfo.getEncoderSettings());
+    modified |= copyIfNotNull(pMergedTrackInfo->ptrISRC(), importedTrackInfo.getISRC());
+    modified |= copyIfNotNull(pMergedTrackInfo->ptrLanguage(), importedTrackInfo.getLanguage());
+    modified |= copyIfNotNull(pMergedTrackInfo->ptrLyricist(), importedTrackInfo.getLyricist());
+    modified |= copyIfNotNull(pMergedTrackInfo->ptrMood(), importedTrackInfo.getMood());
+    modified |= copyIfNotNull(pMergedTrackInfo->ptrMovement(), importedTrackInfo.getMovement());
+    modified |= copyIfNotNull(pMergedTrackInfo->ptrMusicBrainzArtistId(), importedTrackInfo.getMusicBrainzArtistId());
```
Comment (daschuer) [inline src/track/trackrecord.cpp:132]: Unfortunately these line are exceeding the GitHub diff view. Please break them, or use the our GitHub pre-comit hook.

**Comment by uklotzde** [comment]
> Sorry for eating my dog food :-/
> Where do you usually define **EXTRA_METADATA**? For Testing I have put it into util/macros.h Just building.

Oh, you are right, now I remember. We have no build flag/feature for this particular define. It is only set implicitly with -DAOIDE=ON (CMake). This might be the reason why the bugs slipped through unnoticed.

File: src/track/trackrecord.cpp (hunk 1 above)
Comment (uklotzde) [reply src/track/trackrecord.cpp]: I tried to keep the formatting to reduce the risk for merge conflicts. Now exactly that happened during reformatting ;) Fixed.

**Review by daschuer — APPROVED** [review]
Thank  you.

**Comment by daschuer** [comment]
Now it builds.

**Comment by daschuer** [comment]
I have added https://www.mixxx.org/wiki/doku.php/coding_guidelines#non-const_references
I hope we can tweak it in a way that it acceptable for you as well. Please have a look.

**Pull request merged**